### PR TITLE
feat: add language selector

### LINKED ISSUE #81: Add language picker and "add translations" callout
WebUI automatically detects locale based on request headers, but there is also UI for user to manually change locale.

(there is a bunch of complete translations at Transifex waiting to be used: https://www.transifex.com/ipfs/ipfs-share-files/)

We should also add a "translate this!" prompt to get more translations.

### PR DESCRIPTION
**Pull request by rafaelramalho19**
- Adds the language selector
- Updated language files
- By default it uses the user's operating system language 

image

Closes #81.

### CODE REVIEW

File: src/lib/languages.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "en": {
+    "locale": "en",
+    "nativeName": "English",
+    "englishName": "English"
+  }
+}
```
Comment (lidel) [inline src/lib/languages.json:7]: Where is this used?

File: src/i18n.js [hunk 2]
```diff
@@ -7,9 +7,16 @@ import LanguageDetector from 'i18next-browser-languagedetector'
 import en from 'i18next-icu/locale-data/en'
 import pt from 'i18next-icu/locale-data/pt'
 
+const localeData = [en, pt]
+
+export const localesList = [
+  { value: 'en', name: 'English' },
+  { value: 'pt', name: 'Portuguese' }
+]
```
Comment (lidel) [inline src/i18n.js:15]: I think we want to remove this list and switch to `src/lib/languages.json` which can be generated by `npx -q @olizilla/lol public/locales > src/lib/languages.json

**Review by lidel — CHANGES_REQUESTED** [review]
- see below comment about using `src/lib/languages.json`
- we should run `npx -q @olizilla/lol public/locales > src/lib/languages.json` during the build to ensure we don't have "hidden" locales (in my experience nobody remembers to run it by hand)

**Comment by terichadbourne** [comment]
This looks awesome! Is that globe symbol something that's widely recognized as being a language selection?

File: src/i18n.js (hunk 2 above)
Comment (rafaelramalho19) [reply src/i18n.js]: You're 100% right!

**Comment by rafaelramalho19** [comment]
> This looks awesome! Is that globe symbol something that's widely recognized as being a language selection?

No idea, I just searched for free2use icons, but I guess you can suggest another icon 😄

**Comment by jessicaschilling** [comment]
The globe is adorable but let's use the Material UI translation icon for visual consistency with more web things. Updated in https://github.com/ipfs-shipyard/ipfs-share-files/pull/107/commits/fbe33260a1da16d8c5e642c5f35b018bc8bdcd67 but will do some styling on it too.

**Comment by jessicaschilling** [comment]
Updated the visual styling. @lidel can you please take a final pass?

**Review by jessicaschilling — APPROVED** [review]

**Pull request merged**